dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/domain/Hypervisor.java

```diff
@@ -102,11 +103,25 @@ public static Hypervisor create(String type, String maintenanceStatus, List<Prop
 
       public abstract Builder properties(List<Property> properties);
 
-      public abstract Builder disksSpeed(List<DiskSpeed> properties);
+      public abstract Builder diskSpeeds(List<DiskSpeed> diskSpeeds);
 
-      public abstract Builder cpusSpeed(List<CpuSpeed> properties);
+      public abstract Builder cpuSpeeds(List<CpuSpeed> cpuSpeeds);
+
+      abstract Hypervisor autoBuild();
+
+      abstract List<Property> properties();
+
+      abstract List<DiskSpeed> diskSpeeds();
+
+      abstract List<CpuSpeed> cpuSpeeds();
+
+      public Hypervisor build() {
+         properties(properties() != null ? ImmutableList.copyOf(properties()) : null);
+         diskSpeeds(diskSpeeds() != null ? ImmutableList.copyOf(diskSpeeds()) : null);
+         cpuSpeeds(cpuSpeeds() != null ? ImmutableList.copyOf(cpuSpeeds()) : null);
```

Comment:
For lists not marked `@Nullable` we should return an empty list instead of null (or mark them `@Nullable` if appropriate).